internal/tui/stack.go

```diff
@@ -0,0 +1,79 @@
+package tui
+
+import tea "github.com/charmbracelet/bubbletea"
+
+// ComponentStack is a stack of components.
+type ComponentStack struct {
+	components []Component
+}
+
+// NewComponentStack creates a new component stack.
+func NewComponentStack(initial ...Component) *ComponentStack {
+	return &ComponentStack{
+		components: initial,
+	}
+}
+
+// Push adds a component to the top of the stack.
+func (s *ComponentStack) Push(c Component) {
+	s.components = append(s.components, c)
+}
+
+// Pop removes and returns the top component.
+func (s *ComponentStack) Pop() (Component, bool) {
+	if s.IsEmpty() {
+		return nil, false
+	}
+	index := len(s.components) - 1
+	component := s.components[index]
+	s.components = s.components[:index]
+	return component, true
+}
+
+// Top returns the top component without removing it.
+// It will panic if the stack is empty. The TUI logic should guarantee that
+// the stack is never empty.
+func (s *ComponentStack) Top() Component {
+	return s.components[len(s.components)-1]
+}
```

Comment:
Top() is no longer used outside of this struct, let's remove the function and inline it.